Labyrinth/src/LabyrinthOriginal/GameMaster.java

```diff
@@ -33,19 +33,21 @@ public int tilesides() {
 	 */
 	public void gameStartUp(Integer[] distribution, int tilesWide, int tilesTall) {
 		this.tileBranchDistribution = distribution;
+		
+		//set-up main panel (and board displayer and board):
 		Board b = newBoard(tilesWide, tilesTall);
 		b.addPlayer(new Player(b.tileAt(0,tilesWide-1),Color.yellow)); //tl
 		b.addPlayer(new Player(b.tileAt(tilesWide-1,tilesTall-1),Color.blue)); //tr
 		b.addPlayer(new Player(b.tileAt(tilesWide-1,0),Color.red)); //br
 		b.addPlayer(new Player(b.tileAt(0,0),Color.green)); //bl
-		this.boardDisplayer = new BoardDisplayer(b);
+		this.mainPanel = new MainPanel(new BoardDisplayer(b));
+		//Add playerModes to the board displayer for every player on the board
 		for(Player player : b.getPlayers()) {
-			this.boardDisplayer.addMode(new MovePlayerMode(b,player));
+			this.mainPanel.getBoardDisplayer().addMode(new MovePlayerMode(b,player));
 		}
```

Comment:
Adding modes to the boardDisplayer should probably be handled by the GameMaster without needing multiple method calls like this. It might be good to either:


1. Keep the boardDisplayer field
1. Make a method that initializes a BoardDisplayer around our board, and use that method (instead of `new BoardDisplayer`) as an argument to the MainPanel constructor.
1. Some other thing I haven't thought of

Reply:
What if I make a addMode method in the mainPanel class which adds a mode to the mainPanel's boardDisplayer? I could even make it one method that takes in an argument specifying which type of mode it should add (move player, move row, etc). A simpler (maybe better) alternative would be to just make a method in the mainPanel class which adds a move player mode to the mainPanel's boardDisplayer.

Reply:
I think we need to do some refactoring, so this may be changed soon.